### ISSUE TITLE
[4.4] Delete privacy consent api endpoint not working

### ISSUE THREAD

**Issue by andyforrest** [joomla/joomla-cms#41494]
### Steps to reproduce the issue

1. Enable privacy consent plugin and have user opt in to sites privacy consent
2. Get consent id either through privacy consents dashboard from admin component or from GET privacy/consents api endpont
3. Use api endpont /api/index.php/v1/privacy/consents/{consent_id} to delete the desired privacy consent

### Expected result

- Deleted privacy consent

### Actual result

```JSON
{
    "errors": [
        {
            "code": 500,
            "title": "Internal server error"
        }
    ]
}
```

### System information (as much as possible)

- Macos
- PHP 8.2
- Apache 2.0

**Comment by richard67** [comment]
> iirc there is no way to delete privacy consent from admin, only invalidate but my memory can fail

Reading the documentation https://docs.joomla.org/J3.x:Privacy and https://docs.joomla.org/J3.x:Information_Request_Workflow_in_Privacy_Component is seems @alikon is right, privacy consent can not be removed, but a request for data removal can be created and then completed.

**Comment by andyforrest** [comment]
Appologies, it looks like I had a dyslexia moment and misread the docs! In any case, it appears that the endpoint doesn't  work by not returning any relevant response. As it has been listed as a valid endpoint here: https://docs.joomla.org/J4.x:Joomla_Core_APIs#Privacy

**Comment by alikon** [comment]
i think we should amend the documentation page accordingly based on our findings,
 plus we need to remove the DELETE verb from that endpoint, for the 2nd part i'll do a pr,
 anyone would like to do the docs part ?
@andyforrest thank you for bring this up

**Comment by andyforrest** [comment]
@alikon no trouble! Is it possible there is a mistake writing the docs and it was meant to be an endpoint that invalidates consent?
I can edit the docs if you wish

**Comment by alikon** [comment]
please do it if you can really appreciated

for removing the wrong VERB endpoint please test #41520

**Comment by richard67** [comment]
As fas as I can see, the documentation on https://docs.joomla.org/J4.x:Joomla_Core_APIs#Privacy has been fixed. Thanks @andyforrest for that. Closing the issue as we have a PR to fix it, see #41856 . @andyforrest Could you test it and then mark your test result in the issue tracker by going to https://issues.joomla.org/tracker/joomla-cms/41856 , using the blue "Test this" button at the top left corner, selecting the test result and finally submitting? Thanks in advance.
